### ISSUE TITLE
Clicking on input corrupts the datepicker flow

### ISSUE THREAD

**Issue by milankalinic** [dangrossman/bootstrap-daterangepicker#1343]
### Steps to reproduce
- choose start date ie. "01. september"
- click into the "input" for END date
- click "next month" and try to select "31.december"

### Currently
It resets selector to start date

### Expected
To select the 31. december as end date

### Code that causes problem
```JS
formInputsFocused: function(e) {

    // Highlight the focused input
    this.container.find('input[name="daterangepicker_start"], input[name="daterangepicker_end"]').removeClass('active');
    $(e.target).addClass('active');

    // Set the state such that if the user goes back to using a mouse, 
    // the calendars are aware we're selecting the end of the range, not
    // the start. This allows someone to edit the end of a date range without
    // re-selecting the beginning, by clicking on the end date input then
    // using the calendar.
    var isRight = $(e.target).closest('.calendar').hasClass('right');
    if (isRight) {
        this.endDate = null;
        this.setStartDate(this.startDate.clone());
        this.updateView();
    }

},
```

**Comment by tbwnetwork** [comment]
Hi milan,

i had the same issue. my solution was to comment following part:

`formInputsBlurred: function(e) {

            // this function has one purpose right now: if you tab from the first
            // text input to the second in the UI, the endDate is nulled so that
            // you can click another, but if you tab out without clicking anything
            // or changing the input value, the old endDate should be retained

            /*if (!this.endDate) {
                var val = this.container.find('input[name="daterangepicker_end"]').val();
                var end = moment(val, this.locale.format);
                if (end.isValid()) {
                    this.setEndDate(end);
                    this.updateView();
                }
            }*/

        }`

**Comment by tbwnetwork** [comment]
Hi milan,

i had the same issue. my solution was to comment following part in formInputsBlurred :

formInputsBlurred: function(e) {

           /*if (!this.endDate) {
                var val = this.container.find('input[name="daterangepicker_end"]').val();
                var end = moment(val, this.locale.format);
                if (end.isValid()) {
                    this.setEndDate(end);
                    this.updateView();
                }
            }*/

}

**Comment by dangrossman** [comment]
1. I went to http://www.daterangepicker.com/#ex1

2. I clicked on input

3. I clicked on start date Jan 1

4. I clicked on input for END date

5. I clicked on 31 on the calendar

Result: Jan 1 to 31 selected, not unexpected

**Comment by mfn** [comment]
@dangrossman I think you missed to **change the month** via the arrow indicator.

After clicking into the input for END date and then click the month switch error, the date is already wrong and any click onto a calendar day starts a new range selection

**Comment by dangrossman** [comment]
When you click out of the input you were typing in, whatever text is in the input becomes the end date of your range. Now you have a range fully selected, so clicking on a date should be the start of a new selection. That is also not unexpected.

**Comment by mfn** [comment]
> When you click out of the input you were typing

I wasn't typing anything but I think I got your point about the clicking.

I'll try to better explain what I'm really hoping for your confirmation if that is still expected or not:

1. http://www.daterangepicker.com/#ex1
2. click into the "demo" input, DateRangePicker opens
3. Inside the calendar click on 1st of January 2015<br>The "end date" input receives the blur
4. Click inside the input for the END date<br>But do nothing actually
5. click on the right arrow to select the next month
6. The next month isn't selected but the date shown is received as the end date

At this point it's now like it selected whatever date was as the end date already and any further click on the right arrow to browser further months will already start a new date range.

Basically, once I just click inside the END input (but even when I type nothing), it seems the next action leading to loosing the focus on the END input fortifies the date selection instead of allowing my to continue choose an end date via mouse.

This is (IMHO) the unexpected part ...
